modules/parquet/test/parquet-wasm-loader.spec.ts

```diff
@@ -0,0 +1,81 @@
+import test from 'tape-promise/tape';
+// import {validateLoader} from 'test/common/conformance';
+
+import {ParquetWasmLoader, ParquetWasmWriter} from '@loaders.gl/parquet';
+import {load, encode, setLoaderOptions} from '@loaders.gl/core';
+import {Table, Int32Vector, Utf8Vector, BoolVector, Uint8Vector} from 'apache-arrow';
+import {WASM_SUPPORTED_FILES} from './data/files';
+
+const PARQUET_DIR = '@loaders.gl/parquet/test/data/apache';
+const wasmUrl = 'http://localhost:5000/node_modules/parquet-wasm/arrow1/esm_bg.wasm';
```

Comment:
The whole flow of referencing an external wasm file and using the localhost:5000 override seems close to the flow we have for webworkers. Would it make sense to unify the two? I.e. create a small web worker script that includes the WASM and implements the parse interface? I know this would mean losing the non-worker loader, but how much use is this if need to dynamically load a WASM blob from an external URL?

Reply:
And yes, I think both WASM and web workers need to be instantiated asynchronously, right? I don't know much about packaging web workers. In the geopackage loader (which relies on the WASM `sql.js`) we have a similar process where the WASM is loaded from an external source that the user can configure: https://github.com/visgl/loaders.gl/blob/7acfc31abec9145733efed746cbe6699c7530b38/modules/geopackage/src/lib/parse-geopackage.ts#L120-L131

Reply:
I was more thinking in terms of packaging. A number of loaders have a `build-worker` target in `package.json` which then automatically creates a worker file on the npm and provides a way to use a local version of the worker in unit tests. The idea was that you could create a similar `build-worker` step for parquet-wasm which would abstract away all the WASM loading into that worker

Reply:
@felixpalmer In principle that is a good idea. Are you good at wasm bundling / loading? I have looked into this briefly in the past but overall this seemed like a messy corner of javascript and I didn't find any approach that I got excited about. If someone can demonstrate a convincing solution that seems reasonably simple, generic, portable and future proof I'd be happy to consider it.